Add overview of available Models

VedantKP opened the pull request:
This pull request has efforts taken towards resolving issue [#2339 ](https://github.com/pymor/pymor/issues/2339), wherein an overview of all available models would be added to the `docs/source/mor_methods.md` document. 

The current approach for writing an overview involves:

1. Using the existing grouping in the document and adding available models' overview under the relevant subsection.
2. Each overview subsection consists of: 
             - The theoretical equation of the system represented by the model.
             - A short Python code snippet demonstrating how the model can be initialized.
3. Each section is encapsulated in `details` and `summary` tags to allow expansion and collapse of the information as the reader pleases. This is done to avoid cluttering of the page and allow the user to look at one or more overview sections at a time.

VedantKP commented:
Hello @sdrave,

Based on your comments on the issue, I went through the entire reductors api reference and figured that the existing grouping on the documents page seems fine. Based on this assumption that it is ok to continue with the same grouping, I have written overviews for two models: `LTIModel` and `PHLTIModel`. 
 
Please also clarify the following points:
1.	Do we adhere to the current grouping in the mor_methods.md document?
2.	For certain reductors like pymor.reductors.coercive the fom accepted could be any model that implements the Model interface. In this case basically all models apply. How do we then group them? Do you think specifying LTIModel and PHLTIModel’s overviews in the “LTI System MOR” makes sense as it works under the LTI System MOR, whereas other reductors, e.g. under pymor.reductors.coercive and pymor.reductors.basic, accept any implementation of “Model” so we can group all the remaining models under the Parametric MOR section under the appropriate subheadings?
3.	Do we also talk about the transfer function in the overview as it is indeed an alternative of representing a computational system and e.g. data-driven approaches allow using a TransferFunction or any model with a transfer_function attribute. If yes, then should I include this under the `Data Approximation` section?
4.	Does the current approach of using `details` and `summary` tags seem fine?

Let me know what you think about the above points so that I can continue working on the remaining models.

**Note**: I have provided information keeping in mind that we can’t go overboard with the detailing as this is just an overview. If you feel I need to edit some things out or add some more text, do let me know.

HenKlei commented:
Hi @VedantKP,
thanks for the PR and your work! It is very much appreciated.

You can see the result of your changes to the documentation [here](https://docs.pymor.org/external-pr-2396/mor_methods.html).

As a first remark, I think you would like to see a properly resolved link to the class documentation for `LTIModel` and `PHLTIModel`. I am currently not really sure why this does not work and where the issue is, but it has probably to do with the additional HTML tags. Maybe you can take another look?

VedantKP commented:
Hi @HenKlei,

Thanks for your comment.

I noticed that in one of the previous builds a few other issues were reported like the use of raw HTML tags. To resolve that, I have replaced the HTML tags with the `toggle` directive based on the [sphinx-togglebutton](https://sphinx-togglebutton.readthedocs.io/en/latest/) extension which achieves a similar effect. I have also modified the toml and ci files to ensure it is installed correctly. I resolved the other markdown issues as well. Ideally this should have solved all lint issues but the pre-commit.ci check is still failing. It seems to me that this is because of markdownlint-cli2's hook now. Can you please look into this?

Also, can you please tell me in which job is the link being generated that reflects new changes that were pushed? Just like the [link](https://docs.pymor.org/external-pr-2396/mor_methods.html) you posted in your previous comment. The same link doesn't seem to reflect new changes.

HenKlei commented:
Hi @VedantKP,
the CI pipeline to build the docs failed, you can find the log [here](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/5417672). Maybe this helps already in order to find the issues. This is also why the changes are not reflected in the generated documentation.

VedantKP commented:
Hi @HenKlei,

I have resolved all issues and replaced raw HTML with the `admonition` directive to achieve the dropdown effect. The classes in the overview now link to their documentation as well. 

Can you or @sdrave please review the content of the overview and let me know if it is right so that I can continue working on the rest of the models.

HenKlei commented:
Thanks @VedantKP, we will take a look. And sorry for the delay, the message got lost in my mails...

VedantKP commented:
> Thanks @VedantKP, we will take a look. And sorry for the delay, the message got lost in my mails...

Hi @HenKlei,

Thanks for your reply. As I am a student and we are approaching the end of the Winter Semester, I am pushing all the documentation I have done so far to the branch. I will also be changing the status of the pull request from draft. If there are any changes required after your review, I will surely continue working on them as soon as my exams end 😄

sdrave reviewed (changes requested):
Hi @VedantKP, thanks for your work!

- I think `sphinx-togglebutton` is a nice addition to structure this otherwise very long page (and I prefer this over subpages). However, the rendering looks strange with the white box below the header. Maybe there is a problem with our custom CSS, @HenKlei?
- Regarding the organization of the content, the following structure would make more sense to me:

## ModelA
1. Description
2. How to instantiate ModelA? (code snippet)
3. Available MOR methods
    - method x (code snippet)
    - method y (code snippet)

## ModelB
1. Description
2. How to instantiate ModelB? (code snippet)
3. Available MOR methods
    - method x (code snippet)
    - method z (code snippet)

For all code snippets one could use the togglebutton feature. What do the others think @pymor/main?

HenKlei commented:
Hi @VedantKP,
I just pushed some small changes in our CSS file to your branch, I hope this is fine with you.

@sdrave: The issue seems to be with the mathjax renderings and their size in the case that their visibility is set to `hidden`. I just set `display: none` for the hidden elements now, this seems to work as well. We should check what happens in the rendered version once it is updated to be really sure.

Apart from that I agree with @sdrave's suggestions regarding the structure.
